diff -urN --ex /tmp/refactor-eq-workdir/cleanroom-v2/2254/vendor/github.com/moby/moby/client/request.go /tmp/refactor-eq-workdir/bb-merged-2254/vendor/github.com/moby/moby/client/request.go
--- /tmp/refactor-eq-workdir/cleanroom-v2/2254/vendor/github.com/moby/moby/client/request.go	2026-04-07 06:15:11
+++ /tmp/refactor-eq-workdir/bb-merged-2254/vendor/github.com/moby/moby/client/request.go	2026-04-18 05:42:40
@@ -319,9 +319,7 @@
 
 	if cli.userAgent == nil {
 		// No custom User-Agent set: use the default.
-		if req.Header.Get("User-Agent") == "" {
-			req.Header.Set("User-Agent", defaultUserAgent())
-		}
+		req.Header.Set("User-Agent", defaultUserAgent())
 	} else if *cli.userAgent == "" {
 		// User-Agent set to empty value; remove User-Agent.
 		req.Header.Del("User-Agent")
